File: frontend/src/types/plannedExpense.ts
export type PlannedExpense = {
  id: number;
  planned_date: string;
  year: number;
  month: number;
  cost_center_id: number;
  expense_concept_id: number;
  amount: string;
  supplier: string | null;
  document_number: string | null;
  description: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
  cost_center_code?: string | null;
  cost_center_name?: string | null;
  expense_concept_code?: string | null;
  expense_concept_name?: string | null;
};

export type PlannedExpenseCreatePayload = {
  planned_date: string;
  cost_center_id: number;
  expense_concept_id: number;
  amount: string;
  supplier?: string | null;
  document_number?: string | null;
  description?: string | null;
  notes?: string | null;
};

export type PlannedExpenseUpdatePayload = {
  planned_date: string;
  cost_center_id: number;
  expense_concept_id: number;
  amount: string;
  supplier?: string | null;
  document_number?: string | null;
  description?: string | null;
  notes?: string | null;
};

File: backend/app/schemas/planned_expense.py
"""Esquemas Pydantic v2 para gastos planificados."""
from __future__ import annotations

from datetime import date, datetime
from typing import Annotated

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.validation import (
    PlannedMoneyAmount,
    PositiveId,
)

# Tipos anotados con restricciones de longitud
_Supplier = Annotated[str | None, Field(default=None, max_length=150)]
_DocumentNumber = Annotated[str | None, Field(default=None, max_length=80)]


class PlannedExpenseCreate(BaseModel):
    planned_date: date
    cost_center_id: PositiveId
    expense_concept_id: PositiveId
    amount: PlannedMoneyAmount
    supplier: _Supplier = None
    document_number: _DocumentNumber = None
    description: str | None = None
    notes: str | None = None


class PlannedExpenseUpdate(BaseModel):
    planned_date: date
    cost_center_id: PositiveId
    expense_concept_id: PositiveId
    amount: PlannedMoneyAmount
    supplier: _Supplier = None
    document_number: _DocumentNumber = None
    description: str | None = None
    notes: str | None = None


class PlannedExpensePatch(BaseModel):
    planned_date: date | None = None
    cost_center_id: PositiveId | None = None
    expense_concept_id: PositiveId | None = None
    amount: PlannedMoneyAmount | None = None
    supplier: _Supplier = None
    document_number: _DocumentNumber = None
    description: str | None = None
    notes: str | None = None


class PlannedExpenseRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    planned_date: date
    year: int
    month: int
    cost_center_id: int
    expense_concept_id: int
    amount: PlannedMoneyAmount
    supplier: str | None
    document_number: str | None
    description: str | None
    notes: str | None
    created_at: datetime
    updated_at: datetime
    # Campos enriquecidos para facilitar el frontend
    cost_center_code: str | None = None
    cost_center_name: str | None = None
    expense_concept_code: str | None = None
    expense_concept_name: str | None = None


class PlannedExpenseDeleteResponse(BaseModel):
    ok: bool
    message: str
    id: int
